### ISSUE TITLE
CMake build fails due to unsigned/signed comparison

### ISSUE THREAD

**Issue by wouterbeek** [facebook/rocksdb#5612]
### Expected behavior

The following to succeed:

```
cmake --build .
```

### Actual behavior

The following error:

```
[ 58%] Building CXX object CMakeFiles/rocksdb-shared.dir/port/port_posix.cc.o
/home/wbeek/tmp/rocksdb/port/port_posix.cc: In function ‘int rocksdb::port::GetMaxOpenFiles()’:
/home/wbeek/tmp/rocksdb/port/port_posix.cc:195:31: error: comparison of integer expressions of different signedness: ‘rlim_t’ {aka ‘long unsigned int’} and ‘int’ [-Werror=sign-compare]
  195 |   if (no_files_limit.rlim_cur >= std::numeric_limits<int>::max()) {
      |       ~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[2]: *** [CMakeFiles/rocksdb-shared.dir/build.make:2962: CMakeFiles/rocksdb-shared.dir/port/port_posix.cc.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:5187: CMakeFiles/rocksdb-shared.dir/all] Error 2
gmake: *** [Makefile:141: all] Error 2
```

### Steps to reproduce the behavior

Run the following (I'm using Fedora 30, GCC 9.1, CMake 3.14):

```
mkdir build && cd build
cmake ..
cmake --build .
```

**Comment by anand1976** [comment]
I don't have the environment to reproduce it. Could you try changing the comparison to ```static_cast<int>(no_files_limit.rlim_cur) >= std::numeric_limits<int>::max()``` and see if it works?

**Comment by wouterbeek** [comment]
Your proposed change silences the warning, but I think it's not correct: an unsigned integer that is larger than the maximum positive integer (2,147,483,647) but smaller than the maximum unsigned integer (4,294,967,295) will be cast to a negative integer.  In other words, the `if` condition will never succeed.

I think all these comparisons with type `T` on the left-hand side should be rewritten to use the maximum number of type `T` on the right-hand side.

**Comment by anand1976** [comment]
That wouldn't work either since we want to limit max open files to 2,147,483,647. 

Another alternative is to ```static_cast``` the rhs to ```unsigned long```.

**Comment by koushiro** [comment]
I'm using Arch Linux, GCC 9.1.0, CMake 3.15.0.
The same problem :(

**Comment by koushiro** [comment]
Has been fixed in #5651

**Comment by wouterbeek** [comment]
Thnaks for fixing!
